yaml/shared/src/main/scala/org/virtuslab/yaml/internal/load/parse/Parser.scala

```diff
@@ -2,5 +2,7 @@ package org.virtuslab.yaml.internal.load.parse
 
 import org.virtuslab.yaml.YamlError
 import org.virtuslab.yaml.internal.load.reader.Tokenizer
+
 trait Parser:
-  def getEvents(in: Tokenizer): Either[YamlError, Seq[Event]]
+  def getEvents(): Either[YamlError, Seq[Event]]
```

Comment:
This means I can call getEvents and then getNextEvent. You should break it down into two traits.